core/dive.c

```diff
@@ -1925,18 +1950,23 @@ static void merge_samples(struct divecomputer *res, struct divecomputer *a, stru
 
 		/* same-time sample: add a merged sample. Take the non-zero ones */
 		sample = *bs;
+		sample_renumber(&sample, 0, cylinders_map_b);
 		if (as->depth.mm)
 			sample.depth = as->depth;
 		if (as->temperature.mkelvin)
 			sample.temperature = as->temperature;
-		if (as->pressure[0].mbar)
-			sample.pressure[0] = as->pressure[0];
-		if (as->sensor[0])
-			sample.sensor[0] = as->sensor[0];
-		if (as->pressure[1].mbar)
-			sample.pressure[1] = as->pressure[1];
-		if (as->sensor[1])
-			sample.sensor[1] = as->sensor[1];
+		for (j = 0; j < MAX_SENSORS; ++j) {
+			int sensor_id;
+
+			sensor_id = cylinders_map_a[as->sensor[j]];
+			if (sensor_id < 0)
+				continue;
+
+			if (as->pressure[j].mbar)
+				sample.pressure[j] = as->pressure[j];
+			if (as->sensor[j])
+				sample.sensor[j] = sensor_id;
```

Comment:
this looks odd. shouldn't this be sensor_id instead of 'j'? That's the whole point of the map, right?

Reply:
Hmm... I believe this is correct: this transforms `as->sensor[j]` into `sample.sensor[j]`. `sample_renumber()` [which was split off of `dc_cylinder_renumber()`] seems to do the same thing.

Note that `j` loops over MAX_SENSORS (which effectively means that `j` loops over `{0,1}`), not over the cylinders. The way I understand things, a AI divecomputer can measure at most two pressures at the same time, but the cylinder can change...?

But I'm super-tired at the moment, so I might be seeing things wrongly.